src/endpoint.js

```diff
@@ -74,6 +76,19 @@ module.exports = class AbstractEndpoint {
     return this
   }
 
+  //turns on auto-batching for this endpoint
+  enableAutoBatch (interval) {
+    if (this._autoBatch === null) {
+      this._autoBatch = {
+        interval: interval || 1000,
+        set: new Set(),
+        nextBatchPromise: null,
+        autoBatchOverride: false,
```

Comment:
Remove this - it's essentially global state, has problems with race conditions (something might call `many` right before the batch request hits) and it makes it pretty hard to follow what's going on. Instead add an additional parameter like `skipAutoBatch` to `many`/`_many` and pass `true` through when the batch operation calls `many`.

Reply:
yep, makes perfect sense, implemented these recommendations